.github/workflows/ci-pipeline.yml

```
@@ -7,26 +7,26 @@ on:
 
 jobs:
-  generation-tests:
+  build-and-test:
     runs-on: ubuntu-latest
 
     steps:
     - uses: actions/checkout@v3
 
-    - name: Set up Python 3.12.4
+    - name: Set up Python 3.12
       uses: actions/setup-python@v4
```

actionlint (1.7.7)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)